src/guardian/staking-module-guard/staking-module-guard.service.ts

```diff
@@ -30,34 +31,55 @@ export class StakingModuleGuardService {
   private lastContractsStateByModuleId: Record<number, ContractsState | null> =
     {};
 
-  public async getStakingRouterModuleData(
-    stakingRouterModule: SRModule,
-    blockHash: string,
-  ): Promise<StakingModuleData> {
-    const {
-      data: {
-        keys,
-        module: { nonce },
-      },
-    } = await this.stakingRouterService.getStakingModuleUnusedKeys(
-      blockHash,
-      stakingRouterModule,
-    );
+  // public async getStakingRouterModuleData(
+  //   stakingRouterModule: SRModule,
+  //   blockHash: string,
+  // ): Promise<StakingModuleData> {
+  //   const {
+  //     data: {
+  //       keys,
+  //       module: { nonce },
+  //     },
+  //   } = await this.stakingRouterService.getStakingModuleUnusedKeys(
+  //     blockHash,
+  //     stakingRouterModule,
+  //   );
+
+  //   const isDepositsPaused = await this.securityService.isDepositsPaused(
+  //     stakingRouterModule.id,
+  //     {
+  //       blockHash,
+  //     },
+  //   );
+
+  //   return {
+  //     nonce,
+  //     unusedKeys: keys.map((srKey) => srKey.key),
+  //     stakingModuleId: stakingRouterModule.id,
+  //     blockHash,
+  //   };
+  // }
 
-    const isDepositsPaused = await this.securityService.isDepositsPaused(
-      stakingRouterModule.id,
-      {
-        blockHash,
-      },
+  /**
+   * Check vetted among staking modules
+   */
+  public async checkVettedKeysDuplicates(
+    vettedKeys: RegistryKey[],
+    blockData: BlockData,
+  ): Promise<void> {
+    const uniqueKeys = new Set();
+    const duplicatedKeys = vettedKeys.filter(
+      (vettedKey) => uniqueKeys.size === uniqueKeys.add(vettedKey.key).size,
     );
 
-    return {
-      nonce,
-      unusedKeys: keys.map((srKey) => srKey.key),
-      isDepositsPaused,
-      stakingModuleId: stakingRouterModule.id,
-      blockHash,
-    };
+    if (duplicatedKeys.length) {
+      this.logger.warn('Found duplicated vetted key', {
```

Comment:
I would suggest additionally log the module's ID as well

Reply:
in log will be a list of keys with modules, there they were found

Reply:
@eddort unresolved conv